src/main.rs

```diff
@@ -150,11 +165,22 @@ fn main() -> Result<(), String> {
                 let width = (right_x - left_x) as u16;
                 let height = (bottom_y - top_y) as u16;
 
+                let cursor_pt = xcb::Point::new(motion.event_x(), motion.event_y());
+                let result = get_window_at_point(
```

Comment:
The problem with caching is that the user can change the windows at any time (or even the windows themselves can change on their own). My original idea was to query all windows to avoid the slowdown was to query all windows at startup and do the bounds check from within rust, but then the mentioned problem showed up.

Reply:
I did thought about this but it may change so we need to catch other events.

Reply:
The correct way to handle this would be to fetch the tree once and then listen to events to update it cheaply.

Reply:
Yes, that's true.

Reply:
@9ary because it changes the way we currently do some things, it may deserve another PR.

Reply:
As we discussed outside, the tree approach might add lot of complexity. A simpler approach might be (as suggested by novie) to throttle how often we query the tree, and keep working while we wait for the response. X is asynchronous after all. The simplest approach is the one Lonami sent a patch for, but it feels a bit dirty.

Reply:
@expectocode other only approach I can think of is kind of spawn a timer and reset it on every move. When the timer is over, *then* we render. But that sounds a lot more complex.

Reply:
@Lonami Yes, I think it deserves another PR.